pkg/discovery/kv/kv.go

```diff
@@ -63,9 +63,23 @@ func (s *Discovery) Initialize(uris string, heartbeat time.Duration, ttl time.Du
 	s.ttl = ttl
 	s.path = path.Join(s.prefix, discoveryPath)
 
+	var config *store.Config
```

Comment:
I don't think this will be possible with the current structure of libkv and etcd.

When ClientTLS settings are passed, calling through https://github.com/docker/libkv/blob/master/store/etcd/etcd.go#L63 we wind up at https://github.com/coreos/go-etcd/blob/master/etcd/client.go#L115 and then wiring up the TLS at https://github.com/coreos/go-etcd/blob/master/etcd/client.go#L194

However, If I *also* populate the libkv's TLS config option with the client defaults as you suggest (https://github.com/docker/libkv/blob/master/store/store.go#L43) then *after* the transport is established (calls above) libkv's etcd impl clobbers the TLS via https://github.com/docker/libkv/blob/master/store/etcd/etcd.go#L76 and https://github.com/docker/libkv/blob/master/store/etcd/etcd.go#L100  The end result is connections fail with something along these lines:

time="2015-10-01T17:03:00.776288957Z" level=error msg="Registering as \"192.168.122.223:12376\" in discovery failed: Put https://192.168.122.223:12379/v2/keys/docker/nodes/192.168.122.223:12376: x509: certificate signed by unknown authority"